lib/get-commits.js

```diff
@@ -9,16 +9,28 @@ const debug = require('debug')('semantic-release:get-commits');
  *
  * @return {Promise<Array<Object>>} The list of commits on the branch `branch` since the last release.
  */
-module.exports = async ({cwd, env, lastRelease: {gitHead}, logger}) => {
+module.exports = async ({cwd, env, lastRelease: {gitHead}, logger, options: {commitPaths = []} = {}}) => {
   if (gitHead) {
     debug('Use gitHead: %s', gitHead);
   } else {
     logger.log('No previous release found, retrieving all commits');
   }
 
-  Object.assign(gitLogParser.fields, {hash: 'H', message: 'B', gitTags: 'd', committerDate: {key: 'ci', type: Date}});
+  if (commitPaths) {
+    logger.log('Commits are being filtered by the commitPaths property: %s', ...commitPaths);
+  }
+
+  Object.assign(gitLogParser.fields, {
+    hash: 'H',
+    message: 'B',
+    gitTags: 'd',
+    committerDate: {key: 'ci', type: Date},
+  });
   const commits = (await getStream.array(
-    gitLogParser.parse({_: `${gitHead ? gitHead + '..' : ''}HEAD`}, {cwd, env: {...process.env, ...env}})
+    gitLogParser.parse(
+      {_: [`${gitHead ? gitHead + '..' : ''}HEAD`, '--', ...commitPaths]},
```

Comment:
If `commitPaths` is omitted it should not be included here. Otherwise, the default behaviour would change, as only commits in the current directory (set by `cwd`) will be analyzed instead of previously analyzing the whole repository regardless of the path. 
```suggestion
      {_: [`${gitHead ? gitHead + '..' : ''}HEAD`, ...(commitPaths ? ['--', ...commitPaths] : [])]},
```

Reply:
This PR is essentially dead but to answer your question `commitPaths` defaults to an empty array and an empty array is still truthy. The code you suggested will only result in the true side of the ternary so is essentially the same as what was commited in the PR.

Maybe I could write the code in a different way to be clearer and easier to understand.

Reply:
Mhm, yep but with the `'--'` in front and no input afterwards git is defaulting to `./` so we have to check for `length != 0` then.

Reply:
Is that any different from how `git log` without the trailing `--` works?

Reply:
Just tested running `git log --` in the root of my project has the same results as running `git log --` in my test folder (which has much different set of commits

Reply:
Oh, ok. Then I misread my own comment on the other issue. So its even easier to implement :)